x-pack/plugins/cross_cluster_replication/server/lib/autofollow_pattern_serialization.js

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const deserializeAutofollowPattern = (
+  name,
+  { remote_cluster, leader_index_patterns, follow_index_pattern } = {} // eslint-disable-line camelcase
+) => ({
+  name,
+  remoteCluster: remote_cluster,
+  leaderIndexPatterns: leader_index_patterns,
+  followIndexPattern: follow_index_pattern,
+});
+
+export const deserializeListAutofollowPatterns = autofollowPatterns =>
+  Object.entries(autofollowPatterns).reduce(
+    (deSerialized, [name, autofollowPattern]) => ({
+      ...deSerialized,
```

Comment:
It is just for code style, to avoid the `return`statement. True that in a long list with multiple request/seq this would not be efficient but in our case it would not make any difference